daemon/src/db.rs

```diff
@@ -324,21 +366,235 @@ impl Connection {
+    where
+        C: ClosedCfdAggregate,
+    {
+        let mut conn = self.inner.acquire().await?;
+
+        let cfd = sqlx::query!(
+            r#"
+            SELECT
+                uuid as "uuid: model::OrderId",
+                position as "position: model::Position",
+                initial_price as "initial_price: model::Price",
+                taker_leverage as "taker_leverage: model::Leverage",
+                n_contracts as "n_contracts: model::Contracts",
+                counterparty_network_identity as "counterparty_network_identity: model::Identity",
+                role as "role: model::Role",
+                fees as "fees: model::Fees",
+                expiry_timestamp,
+                lock_txid as "lock_txid: model::Txid",
+                lock_dlc_vout as "lock_dlc_vout: model::Vout"
+            FROM
+                closed_cfds
+            WHERE
+                closed_cfds.uuid = $1
+            "#,
+            id
+        )
+        .fetch_one(&mut conn)
+        .await?;
+
+        let expiry_timestamp = OffsetDateTime::from_unix_timestamp(cfd.expiry_timestamp)?;
+
+        let collaborative_settlement = load_collaborative_settlement_tx(&mut conn, id).await?;
+
+        let commit = load_commit_tx(&mut conn, id).await?;
+
+        let non_collaborative_settlement = load_cet(&mut conn, id).await?;
+
+        let refund = load_refund_tx(&mut conn, id).await?;
+
+        let settlement = match (
+            collaborative_settlement,
+            commit,
+            non_collaborative_settlement,
+            refund,
+        ) {
+            (
+                Some(CollaborativeSettlement {
+                    txid,
+                    vout,
+                    payout,
+                    price,
+                    ..
+                }),
+                None,
+                None,
+                None,
+            ) => Settlement::Collaborative {
+                txid,
+                vout,
+                payout,
+                price,
+            },
+            (
+                None,
+                Some(commit_txid),
+                Some(Cet {
+                    txid,
+                    vout,
+                    payout,
+                    price,
+                    ..
+                }),
+                None,
+            ) => Settlement::Cet {
+                commit_txid,
+                txid,
+                vout,
+                payout,
+                price,
+            },
+            (
+                None,
+                Some(commit_txid),
+                None,
+                Some(Refund {
+                    txid, vout, payout, ..
+                }),
+            ) => Settlement::Refund {
+                commit_txid,
+                txid,
+                vout,
+                payout,
+            },
+            _ => {
+                bail!(
+                    "Closed CFD has insane combination of transactions:
```

Comment:
I hope we'll never see that 😅
(Famous last words...)

Reply:
It is not `unreachable`, but I also hope we don't see such an error. The good thing is that the error would not break anything. The CFD would remain in the `cfds` table.